srw_detector.py

```diff
@@ -66,10 +71,11 @@ def trigger(self):
         sim_id = self._sim_id
         sb = SirepoBluesky(self._sirepo_server)
         data = sb.auth('srw', sim_id)
-        aperture = sb.find_element(data['models']['beamline'], 'title', 'Aperture')
-        aperture['horizontalSize'] = x * 1000
-        aperture['verticalSize'] = y * 1000
-        watch = sb.find_element(data['models']['beamline'], 'title', 'Watchpoint')
+        element = sb.find_element(data['models']['beamline'], 'title', self.cname)
+        element[self.spec_name1] = x * 1000
+        element[self.spec_name2] = y * 1000
+        watch = sb.find_element(data['models']['beamline'], 'title', self.watch_id)
+        watch[self._field0] = x
```

Comment:
I don't think it's right. You need to apply changing parameters to the optical element (formerly, `aperture`). You need the watchpoint only to know which report to simulate.

Reply:
Perhaps I am misunderstanding, but as of now the value of ```element``` is only changed which would be like the ```aperture```, the user selects what they want to change however so it could be a different parameter. The watchpoint does not change it just gets set by the user before the scan so that if there are multiple watchpoints they select one. This also makes it so if a watchpoint has a custom name it can be selected.

Reply:
I don't understand here why x is assigned to both the element and watch. Changing parameter(s) should be applied to the element in this case. To select the name of the watch point we will need a separate parameter.

Reply:
I see what you mean now, this has been fixed I am not sure why it was there in the first place.